pnet_sys/src/lib.rs

```diff
@@ -142,7 +142,7 @@ fn ntohs(u: u16) -> u16 {
 }
 
 fn make_in6_addr(segments: [u16; 8]) -> In6Addr {
-    let mut val: In6Addr = unsafe { mem::uninitialized() };
+    let mut val: In6Addr = unsafe { mem::MaybeUninit::<In6Addr>::uninit().assume_init() };
     val.s6_addr = unsafe {
         mem::transmute([htons(segments[0]),
                         htons(segments[1]),
```

Comment:
`MaybeUninit::uninit().assume_init()` is literally equivalent to `mem::uninitialized()`. In fact, [`mem::uninitialized()` is currently implemented as](https://doc.rust-lang.org/nightly/src/core/mem/mod.rs.html#661-667)

```rust
intrinsics::assert_uninit_valid::<T>(); // this panics iff T has invalid bit patterns (e.g. bool, most enums)
MaybeUninit::uninit().assume_init()
```

Instead, since `In6Addr` aka [`libc::in6_addr`](https://docs.rs/libc/0.2.72/libc/struct.in6_addr.html) is `#[repr(C)]`, couldn't this be the following?

```rust
unsafe {
    mem::transmute::<[u16; 8], In6Addr>([
        // ...
    ])
}
```